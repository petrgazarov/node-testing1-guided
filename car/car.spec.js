const Car = require('./car')

function foo() {
  return 5
}
function bar() {
  return {}
}

describe('Car class', () => {
  let celica
  beforeEach(() => {
    celica = new Car()
  })

  it('exists', () => {
    expect(Car).toBeDefined()
  })
  it('Car can create a car instance', () => {
    expect(celica).toBeInstanceOf(Car)
  })
  it('creates cars with a brand property', () => {
    expect(celica).toHaveProperty('brand')
  })
})

describe('our first tests', () => {
  it('foo returns five', () => {
    expect(foo).toBeDefined() // assertion
    expect(foo).toBeTruthy()  // another assertion
    expect(foo()).toBe(5)     // should be related to the ONE thing
  })
  test('bar works', () => {
    expect(bar).toBeDefined()
    expect(bar).toBeTruthy()
    expect(bar()).not.toBe({}) // only if truly same object passes
    expect(bar()).toEqual({})  // keys and values the same
    expect(bar()).toMatchObject({})
  })
})
