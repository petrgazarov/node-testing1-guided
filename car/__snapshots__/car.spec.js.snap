// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car class creates cars with make and model 1`] = `
Car {
  "make": "toyota",
  "model": "celica",
  "owner": "Dan",
}
`;
